src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -320,24 +320,10 @@ struct MapStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::M
      */
     void Set(uint32_t aMapId, const CharSpan & aMapName)
     {
-        mapID = aMapId;
-
-        if (aMapName.empty())
-        {
-            name = CharSpan(mMapNameBuffer, 0);
-        }
-        else if (aMapName.size() > sizeof(mMapNameBuffer))
-        {
-            // Save the truncated name that fits into available size.
-            memcpy(mMapNameBuffer, aMapName.data(), sizeof(mMapNameBuffer));
-            name = CharSpan(mMapNameBuffer, sizeof(mMapNameBuffer));
-        }
-        else
-        {
-            // Save full name.
-            memcpy(mMapNameBuffer, aMapName.data(), aMapName.size());
-            name = CharSpan(mMapNameBuffer, aMapName.size());
-        }
+        mapID            = aMapId;
+        auto mapNameSpan = MutableCharSpan(mMapNameBuffer, kMapNameMaxSize);
+        CopyCharSpanToMutableCharSpan(aMapName, mapNameSpan);
```

Comment:
So this has different behavior from the old code, right?  The old code copied as much as fit.  `CopyCharSpanToMutableCharSpan` will copy nothing in that case, return error, and `mapNameSpan` will be holding random data....

I think the code above also has that problem.

Reply:
Thanks for spotting this @bzbarsky-apple. I did not check the implementation of this method. @cecille, ideally this method does not error as otherwise setting these values by simply using consecutive `.Set` methods would be cumbersome. Hence, I think we should revert to the old implementation.